networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttestationValidator.java

```diff
@@ -191,6 +191,21 @@ private InternalValidationResult singleAttestationChecks(final Attestation attes
                 return REJECT;
               }
 
+              // The current finalized_checkpoint is an ancestor of the block defined by
+              // aggregate.data.beacon_block_root
+              if (!forkChoiceUtilWrapper
+                  .get_ancestor(
+                      recentChainData.getForkChoiceStrategy().orElseThrow(),
+                      data.getBeacon_block_root(),
+                      compute_start_slot_at_epoch(recentChainData.getFinalizedEpoch()))
+                  .map(
+                      ancestorOfLMDVote ->
+                          ancestorOfLMDVote.equals(
+                              recentChainData.getFinalizedCheckpoint().orElseThrow().getRoot()))
```

Comment:
BTW can't it be the concurrency case when `recentChainData` updates its finalized checkpoint between calls `recentChainData.getFinalizedEpoch()` and `recentChainData.getFinalizedCheckpoint()`? 
May be get `recentChainData.getFinalizedCheckpoint()` once into local var to be certainly safe on this?

Reply:
Good point. Will make the change.

Reply:
Done.